src/coreclr/jit/optimizebools.cpp

```diff
@@ -1720,8 +1721,8 @@ bool Compiler::fgFoldCondToReturnBlock(BasicBlock* block)
         return modified;
     }
 
-    retTrueBb  = block->GetTrueTarget();
-    retFalseBb = block->GetFalseTarget();
+    assert(block->TrueTargetIs(retTrueBb));
```

Comment:
there is `Same here - bail out if the block is no longer BBJ_COND after compacting.` comment with a condition that you can remove I presume

Reply:
I don't think we can remove this: Suppose the true target jumps to the false target, and we compact the false target into the true target. This will make the conditional block degenerate, since it now branches to the same target in either case. `fgReplaceJumpTarget` will detect this, and convert the block into a `BBJ_ALWAYS`, so I think we need to check for this case in the caller.